gcc-8.1 compilation issue after #19 merged

After #19 was merged, we are seeing the following gcc-8.1 compile error:

```console
In file included from /home/kellyt/draco/src/experimental/mdspan:72,
                 from /home/kellyt/draco/src/experimental/test/tst_mdspan.cc:12:
/home/kellyt/draco/src/experimental/__p0009_bits/layout_stride.hpp:206:1: error: embedding a directive within macro arguments is not portable [-Werror]
 #if defined(__INTEL_COMPILER)
 ^
/home/kellyt/draco/src/experimental/__p0009_bits/layout_stride.hpp:209:1: error: embedding a directive within macro arguments is not portable [-Werror]
 #else
 ^
/home/kellyt/draco/src/experimental/__p0009_bits/layout_stride.hpp:211:1: error: embedding a directive within macro arguments is not portable [-Werror]
 #endif
 ^
```

* On a positive note, #19 resolved a fatal build error when we attempt to use intel-17.0.4.

cc: @brryan

--- a/include/experimental/__p0009_bits/layout_stride.hpp
+++ b/include/experimental/__p0009_bits/layout_stride.hpp
@@ -198,19 +198,27 @@ public:
 
   // TODO @proposal-bug @proposal-extension layout stride needs this constructor
   // clang-format off
+#if defined(__INTEL_COMPILER)
+  // Work-around for an ICE. layout_stride won't properly SFINAE with ICC, but oh well
   MDSPAN_FUNCTION_REQUIRES(
     (MDSPAN_INLINE_FUNCTION constexpr explicit),
     layout_stride_impl, (std::experimental::extents<Exts...> const& e), noexcept,
     /* requires */ (
       // remember that this also covers the zero strides case because an && fold on an empty pack is true
-#if defined(__INTEL_COMPILER)
-      // Work-around for an ICE. layout_stride won't properly SFINAE with ICC, but oh well
       true
-#else
-      _MDSPAN_FOLD_AND(Strides != dynamic_extent /* && ... */)
-#endif
     )
-  ) : __base_t(__base_t{__member_pair_t(e, __strides_storage_t())})
+  )
+#else
+  MDSPAN_FUNCTION_REQUIRES(
+    (MDSPAN_INLINE_FUNCTION constexpr explicit),
+    layout_stride_impl, (std::experimental::extents<Exts...> const& e), noexcept,
+    /* requires */ (
+      // remember that this also covers the zero strides case because an && fold on an empty pack is true
+      _MDSPAN_FOLD_AND(Strides != dynamic_extent /* && ... */)
+    )
+  )
+#endif
+  : __base_t(__base_t{__member_pair_t(e, __strides_storage_t())})
   { }
   // clang-format on
 

--- a/include/experimental/__p0009_bits/macros.hpp
+++ b/include/experimental/__p0009_bits/macros.hpp
@@ -211,7 +211,7 @@
 //==============================================================================
 // <editor-fold desc="inline variables"> {{{1
 
-#if _MDSPAN_USE_INLINE_VARIABLES
+#ifdef _MDSPAN_USE_INLINE_VARIABLES
 #  define _MDSPAN_INLINE_VARIABLE inline
 #else
 #  define _MDSPAN_INLINE_VARIABLE
@@ -610,4 +610,3 @@ struct __bools;
 
 // </editor-fold> end Pre-C++14 constexpr }}}1
 //==============================================================================
-

--- a/include/experimental/__p0009_bits/subspan.hpp
+++ b/include/experimental/__p0009_bits/subspan.hpp
@@ -125,7 +125,7 @@ struct preserve_layout_right_analysis : integral_constant<bool, result> {
     // nothing else changes (though if they're true, it doesn't matter)
     encountered_first_all,
     encountered_first_pair
-  >;  
+  >;
 };
 
 template <
@@ -137,7 +137,7 @@ template <
 struct preserve_layout_left_analysis : integral_constant<bool, result> {
   using layout_type_if_preserved = layout_left;
   using encounter_pair = preserve_layout_left_analysis<
-    // Only the left-most slice can be a pair.  If we've encountered anything else, 
+    // Only the left-most slice can be a pair.  If we've encountered anything else,
     // we can't preserve any contiguous layout
     (encountered_first_scalar || encountered_first_all || encountered_first_pair) ? false : result,
     // These change in the expected ways
@@ -228,6 +228,7 @@ struct __assign_op_slice_handler<
   __extents_storage_t __exts;
   __strides_storage_t __strides;
 
+#ifdef __INTEL_COMPILER
 #if __INTEL_COMPILER <= 1800
   MDSPAN_INLINE_FUNCTION constexpr __assign_op_slice_handler(__assign_op_slice_handler&& __other) noexcept
     : __offsets(::std::move(__other.__offsets)), __exts(::std::move(__other.__exts)), __strides(::std::move(__other.__strides))
@@ -239,6 +240,7 @@ struct __assign_op_slice_handler<
   ) noexcept
     : __offsets(::std::move(__o)), __exts(::std::move(__e)), __strides(::std::move(__s))
   { }
+#endif
 #endif
 
 // Don't define this unless we need it; they have a cost to compile

